R/make_states.R

```diff
@@ -0,0 +1,38 @@
+make_state_map_one <- function(state, geometry = TRUE, epsg = alarm_epsg(state)) {
+
+    nd <- geomander::get_alarm(state = state, geometry = geometry, epsg = epsg)
+
+    nd <- nd %>%
+        dplyr::summarise(dplyr::across(tidyselect::starts_with("pop"), sum),
+                  dplyr::across(tidyselect::starts_with("vap"), sum),
+                  dplyr::across(tidyselect::matches("_(dem|rep)_"), sum),
+                  dplyr::across(tidyselect::matches("^a[dr]v_"), sum),
+                  geometry = sf::st_union(geometry))
+
+    id_nrv <- stringr::str_detect(colnames(nd), "arv_")
+    id_ndv <- stringr::str_detect(colnames(nd), "adv_")
+
+    nd <- nd %>%
+        dplyr::mutate(nrv = rowMeans(as.data.frame(nd)[,id_nrv]),
+               ndv = rowMeans(as.data.frame(nd)[,id_ndv]),
+               nrv = round(.data$nrv, 1),
+               ndv = round(.data$ndv, 1)) %>%
+        dplyr::relocate(geometry, .after = .data$ndv)
+
+    suppressWarnings(redist::redist_map(nd, pop_col = .data$pop, ndists = 1, pop_tol = 0.01))
+}
+
+
+make_state_plans_one <- function(state, stats = TRUE) {
+
+    m <- matrix(1, nrow = 1, ncol = 5000)
+    map <- make_state_map_one(state)
+
+    pl <- redist:::new_redist_plans(m, map, algorithm = "Single", wgt = NULL)
```

Comment:
R CMD CHECK leaves a note here because of the use of the non-exported redist:::new_redist_plans function. Do we need to export it or copy the relevant code into this package? @CoryMcCartan

Reply:
We will need to copy the code into the package in some shape or form. We don't need as much stuff as in `redist`, as it won't be exported.

Reply:
Hm can you do `redist::redist_plans` instead? That's the user-facing constructor